lexicon/providers/hostingde.py

```diff
@@ -0,0 +1,255 @@
+"""Module provider for hostingde (Hosting.de)"""
+from __future__ import absolute_import
+
+import json
+import logging
+import time
+import requests
+
+from lexicon.providers.base import Provider as BaseProvider
+
+LOGGER = logging.getLogger(__name__)
+
+NAMESERVER_DOMAINS = ['hosting.de']
+
+# be aware to provide an auth_token
+# LEXICON_HOSTINGDE_AUTH_TOKEN
+
+def provider_parser(subparser):
+    """Return the parser for this provider"""
+    subparser.add_argument(
+        "--auth-token", help="specify api key for authentication")
+
+
+class Provider(BaseProvider):
+    """Provider class for Hosting"""
+
+    def __init__(self, config):
+        super(Provider, self).__init__(config)
+        self.domain_id = None
+        self.api_endpoint = 'https://secure.hosting.de/api/dns/v1/json'
+
+
+    def _authenticate(self):
+
+        response = self._get_zone_config()
+
+        LOGGER.debug('authenticate debug: %s', response)
+        if response == []:
+            raise Exception('Domain {} not found'.format(self.domain))
+
+        self.domain_id = response.get('id', None)
+
+    # Helper
+    def _get_zone_config(self):
+
+        data = ({"filter": {
+            "field": "ZoneName",
+            "value": self.domain
+        }})
+        response = self._request(action='POST', url='/zoneConfigsFind', data=data)
+        if response != []:
+            return response[0]
+
+        return []
+
+    # Normal Behaviour List all records. If filters are provided, send to the API if possible,
+    # else apply filter locally. Return value should be a list of records.
+    # Return an empty list if no records found
+    # type, name and content are used to filter records.
+    # caused the fact, that provider will enclose TXT values with ""
+    # we have filter content afterwords
+    def _list_records(self, rtype=None, name=None, content=None):
+        data = {}
+        subfilter = []  # used by API to filter
+        subfilter.append({"field": "zoneConfigId", "value": self.domain_id})
+        if rtype:
+            subfilter.append({"field": "RecordType", "value": rtype})
+        if name:
+            subfilter.append({"field": "RecordName", "value": self._full_name(name)})
+        if subfilter:
+            data.update(
+                {"filter":{"subFilterConnective": "AND", "subFilter": subfilter}})
+
+        LOGGER.debug('list_records filter: %s', data)
+        raw_records = self._request(action='POST', url='/recordsFind', data=data)
+
+        processed_records = []
+
+        for record in raw_records:
+            processed_record = {
+                'type': record['type'],
+                'name': self._full_name(record['name']),
+                'ttl': record['ttl'],
+                'id': record['id'],
+                'content': record['content'][1:-1] if record['type'] == 'TXT'
```

Comment:
You can reuse the `_clean_TXT_record` method defined by superclass `BaseProvider` here.